code/scripting/api/libs/mission.cpp

```diff
@@ -83,10 +83,47 @@ extern TIMESTAMP Game_shudder_time;
 extern int Game_shudder_total;
 extern float Game_shudder_intensity;
 
+constexpr int COUNT_OBJECTS = -1000;
+
+// Returns the indexth object subclass
+template <typename A>
+int object_subclass_at_index(A& object_subclass_array, size_t array_size, int index)
+{
+	int count = 1;
+
+	for (size_t i = 0; i < array_size; ++i)
+	{
+		int objnum = object_subclass_array[i].objnum;
+		if (objnum < 0 || objnum >= MAX_OBJECTS)
+			continue;
+		if (Objects[objnum].flags[Object::Object_Flags::Should_be_dead])
+			continue;
+
+		if (count == index)
+		{
+			return object_subclass_array[i].objnum;
+		}
+
+		++count;
```

Comment:
Won't this mean that, when counting objects, the returned count will always be 1 higher than the actual number of objects?